packages/block-editor/src/components/block-edit/edit.js

```diff
@@ -48,27 +56,225 @@ const Edit = ( props ) => {
 const EditWithFilters = withFilters( 'editor.BlockEdit' )( Edit );
 
 const EditWithGeneratedProps = ( props ) => {
-	const { attributes = {}, name } = props;
+	const { name, clientId, attributes, setAttributes } = props;
+	const hasPatternOverridesDefaultBinding =
+		attributes?.metadata?.bindings?.[ DEFAULT_ATTRIBUTE ]?.source ===
+		'core/pattern-overrides';
+	const registry = useRegistry();
 	const blockType = getBlockType( name );
 	const blockContext = useContext( BlockContext );
+	const registeredSources = getBlockBindingsSources();
+
+	const blockBindings = useMemo(
+		() =>
+			replacePatternOverrideDefaultBindings(
+				name,
+				attributes?.metadata?.bindings
+			),
+		[ attributes?.metadata?.bindings, name ]
+	);
 
 	// Assign context values using the block type's declared context needs.
 	const context = useMemo( () => {
-		return blockType && blockType.usesContext
-			? Object.fromEntries(
-					Object.entries( blockContext ).filter( ( [ key ] ) =>
-						blockType.usesContext.includes( key )
-					)
-			  )
-			: DEFAULT_BLOCK_CONTEXT;
-	}, [ blockType, blockContext ] );
+		const _context =
+			blockType && blockType.usesContext
+				? Object.fromEntries(
+						Object.entries( blockContext ).filter( ( [ key ] ) =>
+							blockType.usesContext.includes( key )
+						)
+				  )
+				: DEFAULT_BLOCK_CONTEXT;
+		// Add block bindings context.
+		const blockBindingsContext = {};
+		if ( attributes?.metadata?.bindings ) {
+			Object.values( attributes?.metadata?.bindings || {} ).forEach(
+				( binding ) => {
+					registeredSources[ binding?.source ]?.usesContext?.forEach(
+						( key ) => {
+							blockBindingsContext[ key ] = blockContext[ key ];
+						}
+					);
+				}
+			);
+		}
+		return { ..._context, ...blockBindingsContext };
```

Comment:
I see it keeps the existing behavior where the context from the block type is merged with the context requested by every source registered for the block instance. We can keep it this way 👍🏻

Reply:
I thought it could cause some issues if anyone is relying on that, even if it is unintendently. So maybe it is safer to do it in a separate pull request if considered necessary.